### PR TITLE
ci: avoid root-owned /h/.config/gcloud dir

### PR DESCRIPTION
**Pull request by devjgm**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/6416)

### CODE REVIEW

**Review by coryan — APPROVED** [review]

File: ci/cloudbuild/build.sh [hunk 1]
```diff
@@ -231,7 +231,9 @@ if [[ "${DOCKER_FLAG}" = "true" ]]; then
   out_dir="${PROJECT_ROOT}/build-out/${DISTRO_FLAG}-${BUILD_NAME}"
   out_home="${out_dir}/h"
   out_cmake="${out_dir}/cmake-out"
-  mkdir -p "${out_home}" "${out_cmake}"
+  # Create every directory that docker will mount as a volume, otherwise the
+  # docker daemon will create it as a root-owned directory.
+  mkdir -p "${out_cmake}" "${out_home}" "${out_home}/.config/gcloud"
```
Comment (devbww) [inline ci/cloudbuild/build.sh:236]: If you're going to `-p` then there isn't much point in mentioning prefix directories.

Comment (devjgm) [reply ci/cloudbuild/build.sh:236]: I'd rather include all 3 mount points explicitly. I think it makes the intent clearer and the code match the comment.

Comment (devbww) [reply ci/cloudbuild/build.sh:236]: Then I guess it should say `${HOME}/.config/gcloud` to match the code then.

Comment (devjgm) [reply ci/cloudbuild/build.sh:236]: `${HOME}/.config/gcloud` is a different path. The path that will be mounted is down in `build-out/<distro>-<build>/h/.config/gcloud`. Outside of docker, we spell that with `${out_home}`.

Comment (devbww) [reply ci/cloudbuild/build.sh:236]: I understand.  I'm just highlighting that "code match the comment" doesn't really apply.  But OK.

Comment (devjgm) [reply ci/cloudbuild/build.sh:236]: OK. Removed `${out_home}` from the mkdir line. PTAL.

**Review by devbww — APPROVED** [review]

**Pull request merged**